pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -86,6 +87,28 @@ func (p *PdhCounterSet) Initialize(className string) error {
 	return nil
 }
 
+// GetEnglishCounter wraps the PdhAddEnglishCounter call that takes unlocalized counter names (as opposed to the other functions which use PdhAddCounter)
+func GetEnglishCounter(className, counterName, instance string) (PdhSingleInstanceCounterSet, error) {
```

Comment:
So, I don't generally get too pedantic about naming, and the original naming probably isn't the best, but...

This is not a single instance counter.  It's a multi-instance counter, but it's recording a single instance of that counter.
A single instance counter is (for example) `\Memory`, where there's always only one instance of memory, and you can get a variety of stats,
` typeperf -q \memory
\memory\Page Faults/sec
\memory\Available Bytes
\memory\Committed Bytes`
This is being used for single instances of multi-instance counters, for example
`typeperf -q "\processor information"
\processor information(*)\Performance Limit Flags
\processor information(*)\% Performance Limit`
Note the (*), indicating you could get those stats on each individual core/hyper thread
`\processor information(0,0)\% Interrupt Time
\processor information(0,1)\% Interrupt Time
\processor information(0,10)\% Interrupt Time
\processor information(0,11)\% Interrupt Time
\processor information(0,12)\% Interrupt Time
\processor information(0,13)\% Interrupt Time
\processor information(0,14)\% Interrupt Time
\processor information(0,15)\% Interrupt Time
\processor information(0,2)\% Interrupt Time
\processor information(0,3)\% Interrupt Time
\processor information(0,4)\% Interrupt Time
\processor information(0,5)\% Interrupt Time
\processor information(0,6)\% Interrupt Time
\processor information(0,7)\% Interrupt Time
\processor information(0,8)\% Interrupt Time
\processor information(0,9)\% Interrupt Time
\processor information(0,_Total)\% Interrupt Time`
(the number in the parenthesis is NUMA node/processor thread)

Reply:
Yup, that's why I didn't call it "single" nor "multi". This function can actually be used for both (if you pass `null` as instance, it behaves like `GetSingle` otherwise, it behaves like `GetMulti` although you can only query one of the multiple instances). For the CPU check, being able to query one instance of multi-instance counters is enough, but if we want this to replace other multi-instance uses (where we need to query more than on instance) we should probably change the function signature to accept an array. In any case, a single function should work for both single/multi instance counters, since that's how the low-level API we are wrapping works.

Reply:
but it always returns a `PdhSingleInstanceCounterSet`

Reply:
True, what I said might not hold when we add support for multi-instance counters where we want to check more than one instance... I take it back. Maybe we can merge `PdhSingleInstanceCounterSet` and `PdhMultiInstanceCounterSet` but I'm not sure that's better. In any case, for now, this PR is specific for the CPU check.